mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraQuotaLimitDao.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaLimit;
+import org.apache.james.core.quota.QuotaScope;
+import org.apache.james.core.quota.QuotaType;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.querybuilder.delete.Delete;
+import com.datastax.oss.driver.api.querybuilder.insert.Insert;
+import com.datastax.oss.driver.api.querybuilder.select.Select;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class CassandraQuotaLimitDao {
+
+    private final CassandraAsyncExecutor queryExecutor;
+    private final PreparedStatement getQuotaLimitStatement;
+    private final PreparedStatement getQuotaLimitsStatement;
+    private final PreparedStatement setQuotaLimitStatement;
+    private final PreparedStatement deleteQuotaLimitStatement;
+
+    @Inject
+    public CassandraQuotaLimitDao(CqlSession session) {
+        this.queryExecutor = new CassandraAsyncExecutor(session);
+        this.getQuotaLimitStatement = session.prepare(getQuotaLimitStatement().build());
+        this.getQuotaLimitsStatement = session.prepare(getQuotaLimitsStatement().build());
+        this.setQuotaLimitStatement = session.prepare(setQuotaLimitStatement().build());
+        this.deleteQuotaLimitStatement = session.prepare((deleteQuotaLimitStatement().build()));
+    }
+
+    public Mono<QuotaLimit> getQuotaLimit(QuotaLimit.QuotaKey quotaKey) {
+        return queryExecutor.executeSingleRow(getQuotaLimitStatement.bind()
+            .setString(QUOTA_COMPONENT, quotaKey.getQuotaComponent().getValue())
+            .setString(QUOTA_SCOPE, quotaKey.getQuotaScope().getValue())
+            .setString(IDENTIFIER, quotaKey.getIdentifier())
+            .setString(QUOTA_TYPE, quotaKey.getQuotaType().getValue()))
+            .map(row -> convertRowToModel(row));
+    }
+
+    public Flux<QuotaLimit> getQuotaLimits(QuotaLimit.QuotaKey quotaKey) {
+        return queryExecutor.executeRows(getQuotaLimitsStatement.bind()
+            .setString(QUOTA_COMPONENT, quotaKey.getQuotaComponent().getValue())
+            .setString(QUOTA_SCOPE, quotaKey.getQuotaScope().getValue())
+            .setString(IDENTIFIER, quotaKey.getIdentifier()))
+            .map(row -> convertRowToModel(row));
+    }
+
+    public Mono<Void> setQuotaLimit(QuotaLimit quotaLimit) {
+        return queryExecutor.executeVoid(setQuotaLimitStatement.bind()
+            .setString(QUOTA_COMPONENT, quotaLimit.getQuotaKey().getQuotaComponent().getValue())
+            .setString(QUOTA_SCOPE, quotaLimit.getQuotaKey().getQuotaScope().getValue())
+            .setString(IDENTIFIER, quotaLimit.getQuotaKey().getIdentifier())
+            .setString(QUOTA_TYPE, quotaLimit.getQuotaKey().getQuotaType().getValue())
+            .setLong(QUOTA_LIMIT, quotaLimit.getQuotaLimit().orElse(null)));
```

Comment:
Question: In which case do we accept `null` as input for a Quota Limit value?

Reply:
when admin set them to null

Reply:
Ok I see it in the current webadmin API